### PR TITLE
Improve support of CFI flash

### PR DESCRIPTION
**Pull request by jeanthom**
Implement a minimalist CFI flash state machine. This is enough for identifying the NOR flash model.

### CODE REVIEW

File: clicky-core/src/devices/platform/pp/flash.rs [hunk 1]
```diff
@@ -2,11 +2,20 @@ use crate::devices::prelude::*;
 
 use byteorder::{ByteOrder, LittleEndian};
 
+#[derive(PartialEq, Clone, Copy)]
```
Comment (daniel5151) [inline clicky-core/src/devices/platform/pp/flash.rs:5]: I'm not sure you actually need any of these bounds?

File: clicky-core/src/devices/platform/pp/flash.rs [hunk 2]
```diff
@@ -85,14 +97,21 @@ impl Memory for Flash {
...
+                    let offset = offset as usize;
+                    let val = LittleEndian::read_u16(&dump[offset..offset + 2]);
+                    Ok(val)
+                } else {
+                    Err(Unimplemented)
+                }
+                
+            }
+            (CFIState::ReadSoftwareID, 0x0) => return Ok(0x00BF), // Manufacturer ID (SST)
+            (CFIState::ReadSoftwareID, 0x1) => return Ok(0x273F), // Device ID (SST39WF800A)
```
Comment (daniel5151) [inline clicky-core/src/devices/platform/pp/flash.rs:112]:
```suggestion
            (CFIState::ReadSoftwareID, 0x0) => Ok(0x00BF), // Manufacturer ID (SST)
            (CFIState::ReadSoftwareID, 0x1) => Ok(0x273F), // Device ID (SST39WF800A)
```

**Review by daniel5151 — APPROVED** [review]
LGTM, with some minor style comments.

I regret not spending the time way-back-when setting up a proper CI for the project, as it would be good to enforce the use of `cargo clippy` and `rustfmt` in-tree... but i'll hold off on that until I see there's steady activity in-repo to merit going through that initial lift :)

File: clicky-core/src/devices/platform/pp/flash.rs (hunk 1 above)
Comment (jeanthom) [reply clicky-core/src/devices/platform/pp/flash.rs:5]: This is where my lack of experience with Rust shows up :-)

I can't compile without these traits and clippy doesn't bother.

Comment (daniel5151) [reply clicky-core/src/devices/platform/pp/flash.rs:5]: okie, compiler knows best ¯\\\_(ツ)_/¯

**Pull request merged**